packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -141,9 +142,9 @@ export default createRule<
 
           return leftPrecedence < OperatorPrecedence.LeftHandSide;
         }
-        context.report({
-          node: parentNode,
+        checkNullishAndReport(context, parserServices, options, [leftNode], {
           messageId: 'preferOptionalChain',
+          node: parentNode,
```

Comment:
[Testing] Nice to see that `requireNullish` is now supported for `(foo ?? {}).bar`-like code! Could you add few tests to assert this?

Something like...

- ```ts
  declare const foo: string // and string | null
  (foo || {}).toString()
  ```

- ```ts
  declare const foo: string | null
  (foo || 'a' || {}).toString()
  ```


- ```ts
  declare const foo: string // and string | null
  (foo || undefined || {}).toString()
  ```